### PR TITLE
Confidenceintervals

### PR DESCRIPTION
**Pull request by pmineiro**
Adds the Cressie-Read (lambda=-2) estimator and CI.  
Also updated the code of the MLE estimator (aka Cressie-Read with lambda 0).

### CODE REVIEW

File: cressieread.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    def add_example(self, p_log, r, p_pred, count=1):
+        if count > 0:
+            w = p_pred / p_log
+            assert w >= 0, 'Error: negative importance weight'
+
+            self.data.append((count, w, r))
+            self.wmax = max(self.wmax, w)
+            self.wmin = min(self.wmin, w)
+
+    def get_interval(self, alpha=0.05, rmin=0, rmax=1):
```
Comment (cheng-tan) [inline cressieread.py]: Can we move get_interval under class Estimator, since the __init__ and add_example are the same?

**Comment by pmineiro** [comment]
Sounds reasonable to me.  Do you want to go for it?  I have "allow edits from maintainers" enabled.

File: cressieread.py (hunk 1 above)
Comment (pmineiro) [reply cressieread.py]: Sounds reasonable to me. Do you want to go for it? I have "allow edits from maintainers" enabled.

Comment (cheng-tan) [reply cressieread.py]: sure, other than this, is this PR ready to be merged?

Comment (pmineiro) [reply cressieread.py]: Yes afaik.

Comment (cheng-tan) [reply cressieread.py]: @marco-rossi29 can you merge this PR and I'll move get_interval under Estimator with my ips interval PR

Comment (marco-rossi29) [reply cressieread.py]: done

**Pull request merged**